### PR TITLE
ci: GitHub improvements

### PR DESCRIPTION
**Pull request by miquelbeltran**
### Description :memo:
- **Purpose**: Adds multiple GitHub Actions and template improvements.
- **Approach**: Copied them over from `raygun4node` ;)

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**Updates**

- Adds PR template
- Adds Issue templates
- Adds PR Title check 
- Adds `develop` branch check to the existing main workflow

### Test plan :test_tube:

- These templates and config files are not new, but rather already in use in  `raygun4node`

### Author to check :eyeglasses:
- [ ] Project and all contained modules builds successfully
- [ ] Self-/dev-tested 
- [ ] Unit/UI/Automation/Integration tests provided where applicable
- [ ] Code is written to standards
- [ ] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [ ] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

**Comment by miquelbeltran** [comment]
Note: `develop` is not the default branch, we should change that!

**Comment by miquelbeltran** [comment]
I cannot add @sumitramanga to the review :thinking:

File: .github/pr.yml
Comment (TheRealAgentK) [inline .github/pr.yml:1]: This is the GH action for conventional commits, right?

**Review by TheRealAgentK — APPROVED** [review]
LGTM

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -3,9 +3,9 @@ name: CI
 on:
   # Triggers the workflow on push or pull request events but only for the master branch
```
Comment (sumitramanga) [inline .github/workflows/main.yml:4]:
```suggestion
  # Triggers the workflow on push or pull request events but only for the master and develop branch
```

Is this change correct?

**Review by sumitramanga — DISMISSED** [review]
Just one thing otherwise LGTM 🍔

**Comment by sumitramanga** [comment]
@miquelbeltran Got the permissions sorted as well so you can assign me reviews again. That was weird!

**Comment by TheRealAgentK** [comment]
> Note: `develop` is not the default branch, we should change that!

Yup

File: .github/workflows/main.yml (hunk 1 above)
Comment (TheRealAgentK) [reply .github/workflows/main.yml:4]: I think it is, @sumitramanga - we want the conventional commits enforced for changes into `develop` as well.

**Comment by miquelbeltran** [comment]
Re-requested review. Ignore the commit list, the PR is now rebased to `develop-fix` from the PR https://github.com/MindscapeHQ/raygun4flutter/pull/172

Once that other PR is merged, things will look correct.

**Review by sumitramanga — DISMISSED** [review]

**Comment by miquelbeltran** [comment]
Need a new review because `develop` has changed.

**Comment by sumitramanga** [comment]
Code changes LGTM. Upon reflection, the commits section looks very old. Either we squash and merge or create a new PR to show a true representation of this PRl Thoughts @miquelbeltran ?

**Review by sumitramanga — APPROVED** [review]

**Comment by TheRealAgentK** [comment]
> Code changes LGTM. Upon reflection, the commits section looks very old. Either we squash and merge or create a new PR to show a true representation of this PRl Thoughts @miquelbeltran ?

I think this is from branch adjustments (having devlop/master now) - but I'd like @miquelbeltran to confirm that when he comes online later.

**Comment by miquelbeltran** [comment]
that's correct! A squash and merge should clean that up

**Pull request merged**